pramen-py/src/pramen_py/utils/__init__.py

```diff
@@ -139,14 +140,10 @@ def get_or_create_spark_session(
     environment variable PRAMENPY_SPARK_CONFIG
     """
     logger.info("Preparing SparkSession")
-    os.environ.update(
-        {
-            "JAVA_HOME": env.str(
-                "PRAMENPY_SPARK_JAVA_HOME",
-                os.environ.get("JAVA_HOME", ""),
-            )
-        }
-    )
+
+    if env.str("PRAMENPY_SPARK_JAVA_HOME", "") != "":
```

Comment:
Good catch, as we do not need to set JAVA_HOME all the time. Not sure, though, if it is possible do not have the JAVA_HOME variable in the environment at all. 

No need to explicitly check on not equality with """:
```suggestion
    if env.str("PRAMENPY_SPARK_JAVA_HOME", ""):
```

But more pythonic approach would be:
```python
try:
    java_home = env.str(....) = ....
except (check what is a type of exception on missing this var):
    pass
else:
    # update os.environ
```

Reply:
If JAVA_PATH is not defined, the path to Java will be determined by PATH.
But if JAVA_PATH="" (which the old code did), Java fill be searched in the current directory, and always fail.

Although not Pythonic, I prefer the shorter version. But...

@ValeriiKhalimendik it is up to you which version to use. But if you go with @zhukovgreen 's suggestion, I'd recommend to extract these 6 lines of code as a separate function `def set_java_home(env: Env)`.

@zhukovgreen , just for my education, is there a link to Pythonic style you are referring to?

Reply:
No:)